pywbem/_subscription_manager.py

```diff
@@ -1162,26 +1170,28 @@ def _create_destination(self, server_id, dest_url, owned):
         # Validate server_id
         server = self._get_server(server_id)
 
+        svr_name = self._get_systemname(server_id)
+
         # Validate the URL by reconstructing it. Do not allow defaults
         _, _, listener_url = parse_url(dest_url, allow_defaults=False)
 
-        this_host = getfqdn()
         ownership = "owned" if owned else "permanent"
 
-        dest_path = CIMInstanceName(DESTINATION_CLASSNAME,
-                                    namespace=server.interop_ns)
-
         dest_inst = CIMInstance(DESTINATION_CLASSNAME)
-        dest_inst.path = dest_path
         dest_inst['CreationClassName'] = DESTINATION_CLASSNAME
         dest_inst['SystemCreationClassName'] = SYSTEM_CREATION_CLASSNAME
-        dest_inst['SystemName'] = this_host
+        # Issue #2704, dummy set because mock requires the property
+        dest_inst['SystemName'] = svr_name or "dummy"
+
         dest_inst['Name'] = _format(
             'pywbemdestination:{0}:{1}:{2}',
             ownership, self._subscription_manager_id, uuid.uuid4())
         dest_inst['Destination'] = listener_url
 
         if owned:
+            dest_path = self._build_path(dest_inst,
+                                         server.interop_ns,
+                                         server_id)
             for i, inst in enumerate(self._owned_destinations[server_id]):
                 if inst.path == dest_path:
```

Comment:
What if the server side implementation uses an implementation class name for SystemCreationClassName? I think the whole idea to build a path and then compare it exactly is not flexiblke enough and should be changed to compare only the key properties we know we can compare. I.e. introduce a `_match_path()` method. Also, we have the systemname already in this method, so we do not need to create it from the server_id on each loop turn.

Reply:
TODO:   Notet that there is also an instance comparison so the same applies to that test.

Reply:
outdated, see my newer comment.

Reply:
Not sure where newer comment is.   This one ties to the issue of finding owned filters, etc.  I did not do anything more with it so far.

Reply:
The newer comment is https://github.com/pywbem/pywbem/pull/2702#discussion_r646192446.